packages/loot-core/src/server/accounts/sync.ts

```diff
@@ -620,6 +646,10 @@ export async function syncAccount(
   );
 
   const acctRow = await db.select('accounts', id);
+  // If syncing an account from Go Cardless it must not use strictIdChecking. This allows
+  // the fuzzy search to match transactions where the import IDs are different. It is a known quirk
+  // that Go Cardless can give two different transaction IDs even though it's the same transaction.
+  const useStrictIdChecking = acctRow.account_sync_source !== 'goCardless';
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏shouldn't this also exclude SimpleFIN?

Reply:
I'm not sure. I was fixing this based on the problems described in [this issue](https://github.com/actualbudget/actual/issues/2877). If Simple FIN is also affected in the same way as GoCardless I can add it in for that too. Please just let me know 🙏🏼

Reply:
Yea, I think the same thing should be applicable to SimpleFIN (and all future bank-sync providers). So I think the condition can be changed to 
```
const useStrictIdChecking = !acctRow.account_sync_source;
```